### PR TITLE
OS-8442 OpenSSL to 3.0.8, backports to 1.0.2

### PR DESCRIPTION
**Pull request by danmcd**
This is currently the 3.0.8 changes only. There may be 1.0.2 changes as well.

### CODE REVIEW

File: openssl3/tools/checksyms.bash [hunk 1]
```diff
@@ -28,6 +28,7 @@ ALLOWED_SYMS="$ALLOWED_SYMS _init _lib_version _PROCEDURE_LINKAGE_TABLE_"
 # Because we let OpenSSL 3 do the library renaming, it includes this in the
 # symbol list. OpenSSL 3 will be adding to this as it adds new symbols.
 ALLOWED_SYMS="$ALLOWED_SYMS OPENSSL_SMARTOS_3.0.0 OPENSSL_SMARTOS_3.0.3"
+ALLOWED_SYMS="$ALLOWED_SYMS OPENSSL_SMARTOS_3.0.8"
```
Comment (bahamat) [inline openssl3/tools/checksyms.bash:31]: We don't need `OPENSSL_SMARTOS_3.0.0` anymore?

Comment (danmcd) [reply openssl3/tools/checksyms.bash:31]: We actually do:
```
smartos-build(~/OS-8442)[0]% nm proto/lib/libcrypto-smartos.so | grep OPENSSL_SMARTOS
[20122] |         0|         0|OBJT |GLOB |3    |ABS    |OPENSSL_SMARTOS_3.0.0
[20126] |         0|         0|OBJT |GLOB |3    |ABS    |OPENSSL_SMARTOS_3.0.3
[20134] |         0|         0|OBJT |GLOB |3    |ABS    |OPENSSL_SMARTOS_3.0.8
smartos-build(~/OS-8442)[0]% 
```

Comment (bahamat) [reply openssl3/tools/checksyms.bash:31]: I don’t understand why it was removed from this line then.

Comment (danmcd) [reply openssl3/tools/checksyms.bash:31]: The line wasn't removed.  This is an ADDITIONAL LINE (see how it still includes the prior value of ALLOWED_SYMS inside it?).

Comment (danmcd) [reply openssl3/tools/checksyms.bash:31]: This is a bash script so there might be a cleaner way to add items on multiple lines, but given how close we are to branching, maybe next time around?

**Review by bahamat — APPROVED** [review]

**Review by bahamat — APPROVED** [review]
openssl3/tools/checksyms.bash needs copyright updated

**Review by bahamat — APPROVED** [review]
Ship it!

**Pull request merged**